addons/base_import_module/models/ir_module.py

```diff
@@ -280,6 +281,7 @@ def _import_zipfile(self, module_file, force=False, with_demo=False):
                     and file.filename.split('/')[1] in MANIFEST_NAMES
                 ]
                 module_data_files = defaultdict(list)
+                dependencies = defaultdict()
                 for manifest in manifest_files:
```

Comment:
1. `topological_sort` uses a different sorting logic from the normal modules in `module_graph.py`. But I think the imported modules doesn't care about that as long as the result is tologocially sorted
2. The `topology_sort` is deterministic for the same input but may return different results for the equivalent `dependencies` with a different key value items order. Could you double check if `manifest_files` always has the same order?

Reply:
I understand your concern. I checked, and unlike `os.listdir()`, which returns a list in an arbitrary order, `infolist` (used for manifests) returns _a list containing a [ZipInfo](https://docs.python.org/3.10/library/zipfile.html#zipfile.ZipInfo) object for each member of the archive. The objects are **in the same order** as their entries in the actual ZIP file on disk if an existing archive was opened_ ([source](https://docs.python.org/3.10/library/zipfile.html#zipfile.ZipFile.infolist))
Isn't it deterministic enough?

Reply:
Does it depends how you zip those files?
As a simple solution, maybe
```py
manifest_files = sorted(
    (manifest.filename.split('/')[0], file)
    for file in z.infolist()
    if file.filename.count('/') == 1
     and file.filename.split('/')[1] in MANIFEST_NAMES
 )
for mod_name, file in manifest_files:
```

Reply:
Indeed! Updated following your suggestion.
It made me realize that I need to extend the `dependencies` in case there is several manifest files (didn't know it was a supported usecase).
Thanks for your help!